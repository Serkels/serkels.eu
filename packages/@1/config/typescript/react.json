{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React",
  "extends": "@douglasduteil/config...typescript/react.json",
  "compilerOptions": {}
}
