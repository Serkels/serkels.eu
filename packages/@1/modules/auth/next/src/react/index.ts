//

export { SessionProvider, signIn, signOut, useSession } from "next-auth/react";
export { AuthSessionProvider } from "./AuthSessionProvider";
export { useLogin } from "./useLogin";
