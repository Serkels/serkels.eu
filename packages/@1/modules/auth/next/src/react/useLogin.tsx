//

import { useServerAction } from "zsa-react";
import login_action from "./login.action";

//

export function useLogin() {
  return useServerAction(login_action);
}
