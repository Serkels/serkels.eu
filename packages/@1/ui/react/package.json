{
  "name": "@1.ui/react",
  "version": "0.1.0",
  "private": true,
  "sideEffects": false,
  "type": "module",
  "exports": {
    "./tailwind.config.ts": {
      "types": "./tailwind.config.ts",
      "default": "./tailwind.config.ts"
    },
    "./*": {
      "types": "./src/*/index.tsx",
      "default": "./src/*/index.tsx"
    },
    "./*/atom": {
      "types": "./src/*/atom.ts",
      "default": "./src/*/atom.ts"
    },
    "./icons": {
      "types": "./src/icons/index.ts",
      "default": "./src/icons/index.ts"
    },
    "./input/*": {
      "types": "./src/input/*.tsx",
      "default": "./src/input/*.tsx"
    },
    "./form/atom": {
      "types": "./src/form/atom.ts",
      "default": "./src/form/atom.ts"
    },
    "./form/*": {
      "types": "./src/form/*.tsx",
      "default": "./src/form/*.tsx"
    }
  },
  "scripts": {
    "type-check": "tsc --build"
  },
  "dependencies": {
    "@1.modules/core": "workspace:*",
    "@react-hook/merged-ref": "1.3.2",
    "@react-hookz/web": "24.0.4",
    "react-slotify": "0.2.2",
    "date-fns": "3.3.1",
    "formik": "2.4.5",
    "framer-motion": "11.0.5",
    "ts-pattern": "5.0.8",
    "react-aria-components": "1.1.1",
    "tailwind-variants": "0.2.0"
  },
  "devDependencies": {
    "@1.config/prettier": "workspace:*",
    "@1.config/tailwindcss": "workspace:*",
    "@1.config/typescript": "workspace:*",
    "@types/react": "18.2.55",
    "@types/react-dom": "18.2.19",
    "@tanstack/react-query": "4.36.1",
    "prettier": "3.2.5",
    "tailwindcss": "3.3.5",
    "typescript": "5.3.3"
  }
}
