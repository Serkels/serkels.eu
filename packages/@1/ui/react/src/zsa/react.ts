//

export { useServerAction } from "zsa-react";
