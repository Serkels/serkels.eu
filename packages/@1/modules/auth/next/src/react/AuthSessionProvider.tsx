"use client";
import type { Session } from "next-auth";
import { getSession, SessionProvider } from "next-auth/react";
import {
  type PropsWithChildren,
  useCallback,
  useEffect,
  useState,
} from "react";

export function AuthSessionProvider({ children }: PropsWithChildren) {
  console.log("!!!!!!!!!!!");
  const [session, setSession] = useState<Session | null>(null);

  const fetchSession = useCallback(async () => {
    const session = await getSession();
    setSession(session);
  }, [getSession]);

  useEffect(() => {
    fetchSession();
  }, [fetchSession]);

  return <SessionProvider session={session}>{children}</SessionProvider>;
}
