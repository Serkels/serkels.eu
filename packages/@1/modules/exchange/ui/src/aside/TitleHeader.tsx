//

import { format_ago } from "@1.modules/core/date";
import {
  is_active_exchange,
  is_completed_exchange,
  is_expired_exchange,
  type Exchange,
} from "@1.modules/exchange.domain";
import { Time } from "@1.ui/react/time";
import Link from "next/link";
import { match, P } from "ts-pattern";

//

type ContextExchange = Pick<
  Exchange,
  "deals" | "expiry_date" | "id" | "places" | "title" | "updated_at"
>;
export function TitleHeader({ exchange }: { exchange: ContextExchange }) {
  return (
    <div className="flex flex-col">
      <h6 className="line-clamp-2 flex-1 text-xl font-bold">
        {exchange.title}
      </h6>
      <SubTitleHeader exchange={exchange} />
    </div>
  );
}

function SubTitleHeader({ exchange }: { exchange: ContextExchange }) {
  return match({
    exchange,
    is_active: is_active_exchange(exchange),
    is_completed: is_completed_exchange(exchange),
    is_expired: is_expired_exchange(exchange),
  })
    .with(
      {
        is_expired: true,
        exchange: { expiry_date: P.select(P.nonNullable) },
      },
      (date) => (
        <Time className="text-danger" timestamp={date}>
          Échange expiré depuis {format_ago(date)}
        </Time>
      ),
    )
    .with(
      { is_completed: true, exchange: { updated_at: P.select() } },
      (date) => (
        <Time className="text-success" timestamp={date}>
          Échange complet depuis {format_ago(date)}
        </Time>
      ),
    )
    .otherwise(({ exchange: { id } }) => (
      <Link href={`/exchanges/${id}`}>Consulter l'échange</Link>
    ));
}
