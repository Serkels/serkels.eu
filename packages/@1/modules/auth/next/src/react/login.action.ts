"use server";

import { signIn } from "@1.modules/auth.next/auth";
import { startSpan } from "@sentry/core";
import { createServerAction } from "zsa";
import { login_form_schema } from "../schema/login";

//

export default createServerAction()
  .input(login_form_schema)
  .handler(async ({ input: { email } }) => {
    return startSpan(
      { name: "login_action#Login(credentials)", op: "www" },
      async function login_action() {
        // const user = await get_user_by_email(email);
        // if (user) {
        //   throw new Error("User already exists");
        // }

        return await signIn("nodemailer", { email, redirect: false });
      },
    );
  });
