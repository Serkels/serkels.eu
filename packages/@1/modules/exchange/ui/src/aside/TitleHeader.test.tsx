//

import type { Exchange } from "@1.modules/exchange.domain";
import { render } from "@testing-library/react";
import { beforeAll, expect, setSystemTime, test } from "bun:test";
import { TitleHeader } from "./TitleHeader";

//

beforeAll(() => {
  setSystemTime(new Date("2022-11-11"));
});

test("should link to the exchange", () => {
  const exchange = {
    deals: [],
    expiry_date: new Date("2022-11-11"),
    id: "vulfpeck-joe-dart-masterclass",
    title: "Joe Dart Masterclass",
    places: 1,
    updated_at: new Date("2022-11-11"),
  } satisfies Partial<Exchange>;
  const { container } = render(<TitleHeader exchange={exchange} />);
  expect(container.outerHTML).toMatchSnapshot();
});

test("should say the exchange expired a day ago", () => {
  const exchange = {
    deals: [],
    expiry_date: new Date("2022-11-10"),
    id: "vulfpeck-joe-dart-masterclass",
    title: "Joe Dart Masterclass",
    places: 1,
    updated_at: new Date("2022-11-11"),
  } satisfies Partial<Exchange>;
  const { container } = render(<TitleHeader exchange={exchange} />);
  expect(container.outerHTML).toMatchSnapshot();
});

test("should say the exchange is complete depuis 1 mois", () => {
  const exchange = {
    deals: [{ id: "douglas-play-bass" }],
    expiry_date: null,
    id: "vulfpeck-joe-dart-masterclass",
    title: "Joe Dart Masterclass",
    places: 1,
    updated_at: new Date("2022-10-11"),
  } satisfies Partial<Exchange>;
  const { container } = render(<TitleHeader exchange={exchange} />);
  expect(container.outerHTML).toMatchSnapshot();
});
