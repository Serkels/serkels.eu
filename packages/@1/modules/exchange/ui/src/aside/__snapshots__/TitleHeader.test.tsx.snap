// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`should link to the exchange 1`] = `"<div><div class="flex flex-col"><h6 class="line-clamp-2 flex-1 text-xl font-bold">Joe Dart Masterclass</h6><a href="/exchanges/vulfpeck-joe-dart-masterclass">Consulter l'échange</a></div></div>"`;

exports[`should say the exchange expired a day ago 1`] = `"<div><div class="flex flex-col"><h6 class="line-clamp-2 flex-1 text-xl font-bold">Joe Dart Masterclass</h6><time datetime="Thu, 10 Nov 2022 00:00:00 GMT" title="Thu, 10 Nov 2022 00:00:00 GMT" class="text-danger">Échange expiré depuis 1 jour</time></div></div>"`;

exports[`should say the exchange is complete depuis 1 mois 1`] = `"<div><div class="flex flex-col"><h6 class="line-clamp-2 flex-1 text-xl font-bold">Joe Dart Masterclass</h6><time datetime="Tue, 11 Oct 2022 00:00:00 GMT" title="Tue, 11 Oct 2022 00:00:00 GMT" class="text-success">Échange complet depuis environ 1 mois</time></div></div>"`;
